src/generateApplicationFormJson.ts

```diff
@@ -19,41 +26,66 @@ interface ApplicationForm {
   fields: ApplicationFormField[];
 }
 
-interface SpreadsheetAdditionalColumns {
-  funderField: string;
+interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+  dataType: string;
+  createdAt: string;
 }
 
 const args = parse<Args>({
   inputFile: String,
   outputFile: String,
   opportunityId: Number,
+  funder: String,
 });
 
-const psvInput = fs.createReadStream(args.inputFile, 'utf8');
+const csvInput = fs.createReadStream(args.inputFile, 'utf8');
 const jsonOutput = fs.createWriteStream(args.outputFile, 'utf8');
 const { opportunityId } = args;
+const { funder } = args;
 
 let applicationForm: ApplicationForm = {
     opportunityId,
     fields: [],
 }
 
-psvInput.pipe(
-  new CsvReadableStream({
-    parseNumbers: true,
-    parseBooleans: true,
-    trim: true,
-    delimiter: '|',
-    asObject: true,
-  }),
-).on('data', (row: ApplicationFormField & SpreadsheetAdditionalColumns) => {
-  const applicationFormField: ApplicationFormField = {
-    canonicalFieldId: row.canonicalFieldId,
-    position: row.position,
-    label: row.label,
+axios('https://api.philanthropydatacommons.org/canonicalFields',{
+  'method': 'GET',
+  'headers' : {
+    'accept': 'application/json',
+    'x-api-key': process.env.apiKey
   }
-  applicationForm.fields.push(applicationFormField);
-}).on('end', () => {
-  jsonOutput.write(JSON.stringify(applicationForm));
-  jsonOutput.close();
-});
+}).then((response) => {
+  let fields: CanonicalField[] = response.data;
+  csvInput.pipe(
+    new CsvReadableStream({
+      parseNumbers: true,
+      parseBooleans: true,
+      trim: true,
+      allowQuotes: true,
+      asObject: true,
+    }),
+  ).on('data', (row: csvRow) => {
+    const label = funder + ': field label';
+    const id = funder + ': external ID';
+    const pos = funder + ': form position';
+    let field: CanonicalField[] | any;
+    if(row[id] !== ''){
```

Comment:
I didn't add a linter, but the formatting here looks like it could use a space between `if` and `(` as well as after `)` before `{`.